src/hal/src/command/mod.rs

```diff
@@ -19,22 +19,33 @@ use std::borrow::Borrow;
 
 /// Trait indicating how many times a Submit can be submitted.
 pub trait Shot {
-    ///
+    /// DOC TODO
     const FLAGS: CommandBufferFlags;
 }
 /// Indicates a Submit that can only be submitted once.
 pub enum OneShot { }
 impl Shot for OneShot { const FLAGS: CommandBufferFlags = CommandBufferFlags::ONE_TIME_SUBMIT; }
+
 /// Indicates a Submit that can be submitted multiple times.
 pub enum MultiShot { }
 impl Shot for MultiShot { const FLAGS: CommandBufferFlags = CommandBufferFlags::EMPTY; }
 
 /// A trait indicating the level of a command buffer.
 pub trait Level { }
+
 /// Indicates a primary command buffer.
+///
+/// Vulkan describes a primary command buffer as one which can be directly submitted
+/// to a queue, and can execute `Secondary` command buffers.
 pub enum Primary { }
 impl Level for Primary { }
+
 /// Indicates a secondary command buffer.
+///
+/// Vulkan describes a secondary command buffer as one which cannot be directly submitted
+/// to a queue, but can be executed by a primary command buffer.  This allows you to
```

Comment:
"This allows you" - I think needs to be rephrased to avoid "you". Also, there is an extra space before it.